framework/test/driver/BundleTest.cpp

```diff
@@ -815,6 +816,209 @@ void TestBundleGetPropertyKeys()
   US_TEST_CONDITION_REQUIRED(keys.end() != std::find(keys.begin(), keys.end(), Constants::BUNDLE_ACTIVATOR), "Test that property keys contain the expected keys.");
 }
 
+#if defined (US_BUILD_SHARED_LIBS)
+// Test installing a bundle with invalid meta-data in its manifest.
+void TestBundleManifestFailures()
+{
+  auto f = FrameworkFactory().NewFramework();
+  f.Start();
+  auto bc = f.GetBundleContext();
+
+  // throw if manifest.json bundle version key is not a string type
+  US_TEST_FOR_EXCEPTION(std::runtime_error, testing::InstallLib(bc, "TestBundleWithInvalidVersionType"));
+  US_TEST_CONDITION(2 == bc.GetBundles().size(), "Test that an invalid bundle.version type results in not installing the bundle.");
```

Comment:
This is not forward compatible. Since the bundle size is hardcoded, If the main executable is changed to not being a bundle this test fails. I would suggest searching for a bundle with the symbolic name from the list of all bundles.

Reply:
I can't check the bundle name for two of the bundles since they are designed not to have bundle names. I think the best thing to check is a hard count on the # of bundles and make sure that those bundles are named ```system_bundle``` and '''main'''.